src/spend.rs

```diff
@@ -497,6 +502,37 @@ fn derived_desc(
     desc.derive(coin.deriv_index, secp)
 }
 
+/// Get value to use for transaction nLockTime in order to
+/// discourage fee sniping.
+///
+/// The approach follows that taken by Bitcoin Core:
+/// - most of the time, the value returned will be the current
+/// block height, but will randomly be up to 100 blocks earlier.
+/// - if the current tip is more than [`MAX_ANTI_FEE_SNIPING_TIP_AGE_SECS`]
+/// seconds old, a locktime value of 0 will be returned.
+pub fn anti_fee_sniping_locktime(
+    now: Duration,
+    tip_height: u32,
+    tip_time_secs: Option<u32>,
+) -> LockTime {
+    tip_time_secs
+        .map(|tip_time| now.as_secs().saturating_sub(tip_time.into()))
+        .filter(|tip_age| *tip_age <= MAX_ANTI_FEE_SNIPING_TIP_AGE_SECS)
+        .map(|_| {
+            // Randomly set locktime further back using current time as source of randomness.
+            let millis = now.subsec_millis();
```

Comment:
Would there be an advantage to use more precision? Probably truncation, which would be significant for instance between one thousand and one billion (nano).

A quick test
```rust
println!("{} - {} - {}", a.subsec_nanos(), a.subsec_micros(), a.subsec_millis());
```

Gives me:
```
140062106 - 140062 - 140
```

So it's probably use the most precision available?

Reply:
I initially considered using micros, but thought millis might be better as most systems should have at least this level of precision. If using a higher precision, I could make sure the logic is such that in case of truncation, we would set the locktime to the current height.